src/maggma/stores/aws.py

```diff
@@ -12,9 +13,12 @@
 from maggma.core import Sort, Store
 from maggma.utils import grouper, to_isoformat_ceil_ms
 from monty.dev import deprecated
+from monty.json import MontyDecoder
 from monty.msgpack import default as monty_default
 from monty.msgpack import object_hook as monty_object_hook
 
+thread_local = threading.local()
```

Comment:
What's the purpose of this module-level variable?

Also, will this cause a problem if we have more than one `S3Store` ?

Reply:
Oh yeah that should not be module level